test/e2e/deployimage/plugin_cluster_test.go

```diff
@@ -0,0 +1,302 @@
+
+	By("loading the controller docker image into the kind cluster")
+	err = kbc.LoadImageToKindCluster()
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+	By("deploying the controller-manager")
+	cmd := exec.Command("make", "deploy", "IMG="+kbc.ImageName)
+	outputMake, err := kbc.Run(cmd)
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+	By("validating that manager Pod/container(s) are restricted")
+	ExpectWithOffset(1, outputMake).NotTo(ContainSubstring("Warning: would violate PodSecurity"))
+
+	By("validating that the controller-manager pod is running as expected")
+	verifyControllerUp := func() error {
+		// Get pod name
+		podOutput, err := kbc.Kubectl.Get(
+			true,
+			"pods", "-l", "control-plane=controller-manager",
+			"-o", "go-template={{ range .items }}{{ if not .metadata.deletionTimestamp }}{{ .metadata.name }}"+
+				"{{ \"\\n\" }}{{ end }}{{ end }}")
+		ExpectWithOffset(2, err).NotTo(HaveOccurred())
+		podNames := util.GetNonEmptyLines(podOutput)
+		if len(podNames) != 1 {
+			return fmt.Errorf("expect 1 controller pods running, but got %d", len(podNames))
+		}
+		controllerPodName = podNames[0]
+		ExpectWithOffset(2, controllerPodName).Should(ContainSubstring("controller-manager"))
+
+		// Validate pod status
+		status, err := kbc.Kubectl.Get(
+			true,
+			"pods", controllerPodName, "-o", "jsonpath={.status.phase}")
+		ExpectWithOffset(2, err).NotTo(HaveOccurred())
+		if status != "Running" {
+			return fmt.Errorf("controller pod in %s status", status)
+		}
+		return nil
+	}
+	defer func() {
+		out, err := kbc.Kubectl.CommandInNamespace("describe", "all")
+		ExpectWithOffset(1, err).NotTo(HaveOccurred())
+		fmt.Fprintln(GinkgoWriter, out)
+	}()
+	EventuallyWithOffset(1, verifyControllerUp, time.Minute, time.Second).Should(Succeed())
+	By("creating an instance of the CR")
+	sampleFile := filepath.Join("config", "samples",
+		fmt.Sprintf("%s_%s_%s.yaml", kbc.Group, kbc.Version, strings.ToLower(kbc.Kind)))
+
+	sampleFilePath, err := filepath.Abs(filepath.Join(fmt.Sprintf("e2e-%s", kbc.TestSuffix), sampleFile))
+	Expect(err).To(Not(HaveOccurred()))
+
+	EventuallyWithOffset(1, func() error {
+		_, err = kbc.Kubectl.Apply(true, "-f", sampleFilePath)
+		return err
+	}, time.Minute, time.Second).Should(Succeed())
+
+	By("applying the CRD Editor Role")
+	crdEditorRole := filepath.Join("config", "rbac",
+		fmt.Sprintf("%s_editor_role.yaml", strings.ToLower(kbc.Kind)))
+	EventuallyWithOffset(1, func() error {
+		_, err = kbc.Kubectl.Apply(true, "-f", crdEditorRole)
+		return err
+	}, time.Minute, time.Second).Should(Succeed())
+
+	By("applying the CRD Viewer Role")
+	crdViewerRole := filepath.Join("config", "rbac", fmt.Sprintf("%s_viewer_role.yaml", strings.ToLower(kbc.Kind)))
+	EventuallyWithOffset(1, func() error {
+		_, err = kbc.Kubectl.Apply(true, "-f", crdViewerRole)
+		return err
+	}, time.Minute, time.Second).Should(Succeed())
+
+	By("validating that pod(s) status.phase=Running")
```

Comment:
@NikhilSharmaWe since it is working locally:

<img width="1052" alt="Screenshot 2022-06-27 at 17 59 34" src="https://user-images.githubusercontent.com/7708031/176003959-5b1b5076-797e-46ef-b6d8-a5fe898b6476.png">

Let's comment it here and we will check in a follow up why does not work on prow

c/c @varshaprasad96